src/main/java/mate/academy/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.service.shoppingcart;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.model.Book;
+import mate.academy.model.CartItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.repository.BookRepository;
+import mate.academy.repository.CartItemRepository;
+import mate.academy.repository.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Cart not found for user id: " + userId)
+        );
+    }
+
+    @Override
+    public void addBook(Long userId, Long bookId, int quantity) {
+        ShoppingCart cart = getByUserId(userId);
+        Book book = bookRepository.findById(bookId).orElseThrow(
+                () -> new EntityNotFoundException("Book not found: " + bookId)
+        );
+        CartItem cartItem = new CartItem();
```

Comment:
let's get cartItems from cart and check if there is book with the same id, if yes - increase quantity